### PR TITLE
fix: Context data manager

### PR DESCRIPTION
**Pull request by paulomeurerzup**
Signed-off-by: paulomeurerzup <paulo.meurer@zup.com.br>

### Description and Example

This PR solves some issues related to lists inside lists.

### Checklist

Please, check if these important points are met using `[x]`:

- [X] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [X] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [X] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | Classname | Time|
--- | --- | ---|
evaluateContexts_should_get_null_value_from_evaluation | br.com.zup.beagle.android.context.ContextDataManagerTest | 1.688|
evaluateContexts_should_get_value_from_evaluation | br.com.zup.beagle.android.context.ContextDataManagerTest | 1.382|


<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle/src/main/java/br/com/zup/beagle/android/context/ContextDataManager.kt [hunk 1]
```diff
@@ -96,6 +96,10 @@ internal class ContextDataManager(
                 notifyBindingTokens(bindingTokens, contextStack, binding)
             }
             viewBinding.remove(view)
+        } else {
+            view.getContextBinding()?.let {
+                notifyBindingChanges(it)
+            }
```
Comment (matheusribeirozup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/context/ContextDataManager.kt:102]: What do you think of adding integrated unit tests to validate this change?

File: android/beagle/src/main/java/br/com/zup/beagle/android/context/ImplicitContextManager.kt [hunk 2]
```diff
@@ -53,7 +53,7 @@ class ImplicitContextManager {
     ) {
         implicitContextData.forEach { implicitContext ->
             implicitContext.caller.forEach {
-                if (toCompare == it) {
+                if (toCompare === it) {
```
Comment (matheusribeirozup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/context/ImplicitContextManager.kt:56]: The same here

File: android/beagle/src/main/java/br/com/zup/beagle/android/utils/ViewExtensions.kt [hunk 3]
```diff
@@ -77,10 +77,15 @@ internal fun View.getParentContextData(): View? {
...
-    this.setContextBinding(contextBinding)
+    val contextBinding = this.getContextBinding()
+    if (contextBinding != null) {
+        contextBinding.context = normalizedContext
+        contextBinding.cache.evictAll()
+    } else {
+        this.setContextBinding(ContextBinding(
+            normalizedContext
+        ))
+    }
```
Comment (matheusribeirozup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/utils/ViewExtensions.kt:88]: The same here

**Pull request merged**